lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,192 @@
+      container_deployment_nodes.count > 1
+    end
+
+    def ssh_user
+      ssh_auth.userid
+    end
+
+    def add_deployment_provider(options)
+      object_send(:add_deployment_provider, options)
+    end
+
+    def regenerate_ansible_inventory
+      object_send(:generate_ansible_yaml)
+    end
+
+    def regenerate_ansible_subscription_inventory
+      object_send(:generate_ansible_inventory_for_subscription)
+    end
+
+    def add_automation_task(task)
+      ar_method do
+        @object.automation_task = AutomationTask.find_by_id(task.id)
+        @object.save!
+      end
+    end
+
+    def customize(options = {})
+      ar_method do
+        if @object.customizations.empty?
+          @object.customizations = {:agent => {}}
+        end
+        @object.customizations[:agent].merge!(options)
+        @object.save!
+      end
+    end
+
+    def perform_scp(ip, username, local_path, remote_path)
+      Net::SCP.upload!(ip,
+                       username,
+                       local_path,
+                       remote_path,
+                       :ssh => {:key_data => ssh_auth.auth_key})
+    end
+
+    def perform_agent_commands(ip, username, commands = [])
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      ssh.perform_commands(commands)
+    end
+
+    def check_connection(deployment_host_ip, username, nodes_ips)
+      agent = LinuxAdmin::SSHAgent.new(ssh_auth.auth_key, "/tmp/ssh_manageiq/ssh_manageiq_#{id}")
+      success, unreachable_ips = agent.with_service do |socket|
+        ssh = LinuxAdmin::SSH.new(deployment_host_ip, username)
+        unreachable_ips = nodes_ips.select do |sub_ip|
+          ssh.perform_commands(["sudo -E ssh -o 'StrictHostKeyChecking no' #{username}@#{sub_ip} echo 0"], socket)[:stdout].exclude?("0\r\n")
+        end
+        [unreachable_ips.empty?, unreachable_ips]
+      end
+      unless success
+        raise StandardError, "couldn't connect to : #{unreachable_ips.join(',')}"
+      end
+      success
+    end
+
+    def playbook_running?
+      pid = ar_method do
+        @object.customizations[:agent][:ansible_pid] if @object.customizations[:agent]
+      end
+      !pid.nil?
+    end
+
+    def run_playbook_command(ip, username, cmd)
+      result = {:finished => false}
+      ssh = LinuxAdmin::SSH.new(ip, username, ssh_auth.auth_key)
+      ansible_process_pid = ansible_pid
+      if ansible_process_pid
+        process_running = ssh.perform_commands(["sudo kill -0 #{ansible_process_pid}"])[:stdout]
+        unless process_running.empty?
+          result[:finished] = true
+          result[:stdout] = ssh.perform_commands(["sudo cat /tmp/ansible.log"])[:stdout]
+          stop_agent(ssh)
+        end
+      else
+        start_playbook_command(ssh, username, cmd)
+      end
+      result
+    end
+
+    def start_playbook_command(ssh, username, cmd)
+      pid, socket = create_agent(ssh, username)
+      ssh.perform_commands(["sudo rm -f /tmp/*ansible.log*"])
```

Comment:
@alongoldboim once you'll have path constants we'll avoid using wildcards and remove only the specific files.